core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala

```diff
@@ -1619,6 +1619,55 @@ class KRaftClusterTest {
     }
   }
 
+  /**
+   * Test that once a cluster is formatted, a bootstrap.metadata file that contains an unsupported
+   * MetadataVersion is not a problem. This is a regression test for KAFKA-19192.
+   */
+  @Test
+  def testOldBootstrapMetadataFile(): Unit = {
+    val baseDirectory = TestUtils.tempDir().toPath()
+    val cluster = new KafkaClusterTestKit.Builder(
+      new TestKitNodes.Builder().
+        setNumBrokerNodes(1).
+        setNumControllerNodes(1).
+        setBaseDirectory(baseDirectory).
+          build()).
+      setDeleteOnClose(false).
+        build()
+    try {
+      cluster.format()
+      cluster.startup()
+      cluster.waitForReadyBrokers()
+    } finally {
+      cluster.close()
+    }
```

Comment:
Scala recently added the Using type. It has support for closing or releasing resources.

```java
    Using.resource(new KafkaClusterTestKit.Builder(
      new TestKitNodes.Builder().
        setNumBrokerNodes(1).
        setNumControllerNodes(1).
        setBaseDirectory(baseDirectory).
          build()).
      setDeleteOnClose(false).
        build()
    ) { cluster =>
      cluster.format()
      cluster.startup()
      cluster.waitForReadyBrokers()
    }
```

This comment also applies to the creation and usage of the `cluster2` object in this method.